src/Plugin/RulesAction/EntityCreateDeriver.php

```diff
@@ -92,6 +92,7 @@ public function getDerivativeDefinitions($base_plugin_definition) {
         $required = ($field_name == $bundle_key);
         $multiple = ($definition->getCardinality() === 1) ? FALSE : TRUE;
         $this->derivatives["entity:$entity_type_id"]['context'][$field_name] = ContextDefinition::create($definition->getType())
+          ->setAssignmentRestriction(ContextDefinition::ASSIGNMENT_RESTRICTION_INPUT)
```

Comment:
now you are adding the assignment restriction to every context definition, but you should only add it to the bundle context definition.